ksql-rest-app/src/main/java/io/confluent/ksql/api/auth/KsqlAuthorizationFilter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.auth;
+
+import io.confluent.ksql.security.KsqlAuthorizationProvider;
+import io.vertx.core.Handler;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.RoutingContext;
+import java.security.Principal;
+
+public class KsqlAuthorizationFilter implements Handler<RoutingContext> {
+
+  private final WorkerExecutor workerExecutor;
+  private final KsqlAuthorizationProvider ksqlAuthorizationProvider;
+
+  public KsqlAuthorizationFilter(final WorkerExecutor workerExecutor,
+      final KsqlAuthorizationProvider ksqlAuthorizationProvider) {
+    this.workerExecutor = workerExecutor;
+    this.ksqlAuthorizationProvider = ksqlAuthorizationProvider;
+  }
+
+  @Override
+  public void handle(final RoutingContext routingContext) {
+
+    workerExecutor.executeBlocking(
+        promise -> {
+          final User user = routingContext.user();
+          if (user == null) {
+            promise.complete();
```

Comment:
Why does the filter complete successfully if the user is null? That seems like reason to fail since it means no user was authenticated.

Reply:
Authentication come before the authorization here. In the case the user failed authentication then a 401 will be returned and this won't be executed.

If the user passes authentication a User object will be created and set on the routing context.

In the case there is no authentication enabled, the user will be null and there will be no authorization either.

Reply:
> In the case there is no authentication enabled, the user will be null and there will be no authorization either.

But the filter is only set if authentication is enabled, which seems like a sound model (in terms of expected usage of the filter) going forward as well? As in, if I were writing an application and plugged in an authorization module, I'd expect it to fail for an unauthenticated user. Is that not standard?

Reply:
I actually changed this in the next PR :)